src/scripts/app.ts

```diff
@@ -1432,6 +1438,37 @@ export class ComfyApp {
       this.loadTemplateData({ templates })
     }
 
+    // Check workflow first - it should take priority over parameters
+    // when both are present (e.g., in ComfyUI-generated PNGs)
+    if (workflow) {
+      let workflowObj: ComfyWorkflowJSON
+      try {
+        workflowObj =
+          typeof workflow === 'string' ? JSON.parse(workflow) : workflow
+      } catch (err) {
+        console.error('Failed to parse workflow:', err)
+        this.showErrorOnFileLoad(file)
+        return
+      }
+
+      // Validate workflow is a proper object
+      if (
+        !workflowObj ||
+        typeof workflowObj !== 'object' ||
+        Array.isArray(workflowObj)
+      ) {
+        console.error('Invalid workflow structure')
+        this.showErrorOnFileLoad(file)
+        return
+      }
+
+      await this.loadGraphData(workflowObj, true, true, fileName, {
+        openSource
+      })
+      return
+    }
```

Comment:
I'd probably extract a method that returns `ComfyWorkflowJSON|undefined`, but how about this?

```suggestion
    if (workflow) {
      let workflowObj: ComfyWorkflowJSON
      try {
        workflowObj =
          typeof workflow === 'string' ? JSON.parse(workflow) : workflow
      } catch (err) {
        console.error('Failed to parse workflow:', err)
        this.showErrorOnFileLoad(file)
      }

      // Validate workflow is a proper object
      if (
        typeof workflowObj !== 'object' ||
        Array.isArray(workflowObj)
      ) {
        console.error('Invalid workflow structure')
      } else {
        await this.loadGraphData(workflowObj, true, true, fileName, {
          openSource
        })
        return
      }
    }
```